src/hooks/useQuery.ts

```diff
@@ -51,23 +52,23 @@ export const useQuery = <T = any, V = object>(
     (opts?: Partial<OperationContext>) => {
       unsubscribe.current();
 
-      setState(s => ({ ...s, fetching: true }));
+      if (args.pause) {
+        unsubscribe.current = noop;
+        return;
```

Comment:
I think the early `return` here is the reason for the test failures. Basically, `waitForNextUpdate` expects some asynchronous thing to happen (i.e. a `Promise` to be resolved when running the `useEffect`), otherwise it'll never get a signal that the effect completed. You should be able to switch the failing tests to sync tests (remove `async` keyword and `await waitForNextUpdate`) and I'd expect them to pass.

Reply:
Thanks Parker, I'll investigate tonight 👍

Reply:
Looking a little more closely, it appears that the failed tests are related to the effect not re-running when new variables, query, and requestPolicy are received 🤔 Not totally sure why that would be.

Reply:
@parkerziegler I've managed to fix the tests by removing the waitForUpdate call. Given tests themselves are checking that a re-fetch is triggered we can get away without waiting for state updates.

As for why those state updates aren't taking place in a test environment, I suspect it's related to the mock implementation of executeQuery.

Reply:
Gotcha, will dig into this more tomorrow to confirm. But agree, as long as we're sure that `executeQuery` is being called again when a new `query`, `variables`, or `requestPolicy` are received, then I'm not too worried about the state updates.